api/api.py

```diff
@@ -9,99 +9,204 @@
-        webapp2.Route(r'/<:[0-9a-f]{24}>/file',                     sessions.Session, handler_method='file', methods=['POST']),
-        webapp2.Route(r'/<:[0-9a-f]{24}>/file/<:[^/]+>',            sessions.Session, handler_method='file'),
-        webapp2.Route(r'/<:[0-9a-f]{24}>/acquisitions',             acquisitions.Acquisitions, name='acquisitions'),
-    ]),
-    webapp2_extras.routes.PathPrefixRoute(r'/api/acquisitions', [
-        webapp2.Route(r'/count',                                    acquisitions.Acquisitions, handler_method='count', methods=['GET']),
-        webapp2.Route(r'/schema',                                   acquisitions.Acquisition, handler_method='schema', methods=['GET']),
-        webapp2.Route(r'/<:[0-9a-f]{24}>',                          acquisitions.Acquisition, name='acquisition'),
-        webapp2.Route(r'/<:[0-9a-f]{24}>/file',                     acquisitions.Acquisition, handler_method='file', methods=['POST']),
-        webapp2.Route(r'/<:[0-9a-f]{24}>/file/<:[^/]+>',            acquisitions.Acquisition, handler_method='file'),
-    ]),
-    webapp2.Route(r'/api/jobs',                                     jobs.Jobs),
+    webapp2.Route(
+        r'/api/jobs',
+        jobs.Jobs
+    ),
     webapp2_extras.routes.PathPrefixRoute(r'/api/jobs', [
-        webapp2.Route(r'/next',                                     jobs.Jobs, handler_method='next', methods=['GET']),
-        webapp2.Route(r'/count',                                    jobs.Jobs, handler_method='count', methods=['GET']),
-        webapp2.Route(r'/addTestJob',                               jobs.Jobs, handler_method='addTestJob', methods=['GET']),
-        webapp2.Route(r'/<:[^/]+>',                                 jobs.Job,  name='job'),
+        webapp2.Route(
+            r'/next',
+            jobs.Jobs, handler_method='next', methods=['GET']
+        ),
+        webapp2.Route(
+            r'/count',
+            jobs.Jobs, handler_method='count', methods=['GET']
+        ),
+        webapp2.Route(
+            r'/addTestJob',
+            jobs.Jobs, handler_method='addTestJob', methods=['GET']
+        ),
+        webapp2.Route(
+            r'/<:[^/]+>',
+            jobs.Job,  name='job'
+        ),
     ]),
-]
+    webapp2.Route(
+        r'/api/groups',
+        grouphandler.GroupHandler, handler_method='get_all', methods=['GET']
+    ),
+    webapp2.Route(
+        r'/api/groups',
+        grouphandler.GroupHandler, methods=['POST']
+    ),
+    webapp2.Route(
+        _format(r'/api/groups/<_id:{group_id_re}>'),
+        grouphandler.GroupHandler, name='group'
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:groups>/<cid:{group_id_re}>/<list_name:roles>'),
+        listhandler.ListHandler, name='group'
+    ),
+    webapp2.Route(
+        _format(r'/api/<cont_name:groups>/<cid:{group_id_re}>/<list_name:roles>/<site:{site_re}>/<_id:{user_id_re}>'),
```

Comment:
@kofalt  could you give me an example comment for a route (maybe one that you found particularly difficult to understand)?

I definitely want the routing table to be excessively clear, but I would like to avoid too verbose or redundant content.